### DISCUSSION TITLE
How to incorporate Eloquent API resource with QueryBuilder

### DISCUSSION THREAD

**Discussion by wonder95** [spatie/laravel-query-builder#873] [Q&A] [answered]
In reading through the QB docs and other blog posts, I see references to [using Eloquent API resources](https://spatie.be/docs/laravel-query-builder/v5/features/including-relationships#content-eloquent-api-resources), but they are just a reference with no details. What I am trying to understand is how to use a collection for the primary model being returned, not just for a relationship inside that resource.

For instance, here is a query that returns a model called `WorkHours`:
```
 return Inertia::render('Users/WorkHours/Index', [
            'workHours' => QueryBuilder::for(WorkHours::class)
                ->where('user_id', $user->id)
                ->with('area')
                ->allowedIncludes(['area'])
                ->allowedSorts('date')
                ->defaultSort('-date')
                ->allowedFilters([
                    AllowedFilter::callback('year', function(Builder $query, $value) {
                        $thirdSaturdayPrevious = Carbon::parse('Third Saturday of April ' . $value-1)->toDateString();
                        $thirdSaturdayThis = Carbon::parse('Third Saturday of April ' . $value)->toDateString();
                        $query->whereBetween('date', [$thirdSaturdayPrevious, $thirdSaturdayThis]);
                    })
                ])
                ->paginate(20)
                ->withQueryString(),
            'user' => new UserResource($user),
            'areas' => Area::with('approvers')->get()
        ]);
 ```
 Before implementing QueryBuilder, I was using this:
 ```
  'workHours' => WorkHoursResource::collection(WorkHours::where('user_id', $user->id)
                ->where('approved', true)
                ->with(['approver', 'area', 'user'])->get()),
 ```
 and I would do stuff like converting boolean values to `Yes` or `No` in my resource.
 
 What I can't figure out is how to use my `WorkHoursResource` for the returned `WorkHours` items. I can't wrap the whole item, since that returns all of the pagination meta data, and I've tried using `WorkHoursResource` at various places in the query, but no dice.
 
 Is it even possible to use a resource for the main QueryBuilder item?

**Comment by KnudH** [comment, marked as answer]
I'm not sure, if you still need help with this, but will give a try. Overall, it is stupid simple to incorporate both, as the QueryBuilder is doing nothing more than creating the query. So you can just take the query build from QueryBuilder, execute it and give the result to your Eloquent API Resource / Collection. 

```
return \App\Http\Resources\PersonResource::collection(
    QueryBuilder::for(Person::class)
        ->allowedFilters(['first_name', 'last_name'])
        ->allowedSorts(['first_name', 'last_name'])
        ->paginate()
);
```

**Comment by wonder95** [comment]
Yes, I have figured it out since then.

Thanks.
